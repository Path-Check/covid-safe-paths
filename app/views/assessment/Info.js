import React from 'react';
import {
  ImageBackground,
  SafeAreaView,
  ScrollView,
  StyleSheet,
  View,
} from 'react-native';
import { SvgXml } from 'react-native-svg';

/**
 * @typedef { import("react").ReactNode } ReactNode
 */

/** @type {React.FunctionComponent<{
 *   ctaAction?: () => void;
 *   ctaColor?: string;
 *   ctaTitle?: string;
 *   description?: ReactNode;
 *   footer?: ReactNode;
 *   image: string;
 *   title: string;
 * }>} */
export const Info = ({
  backgroundColor,
  backgroundImage,
  children,
  scrollStyle,
  footer,
  icon,
}) => {
  return (
    <SafeAreaView
      backgroundColor={backgroundColor}
      style={assessmentStyles.container}>
      <ImageBackground
        source={backgroundImage}
        style={assessmentStyles.backgroundImage}>
        <ScrollView style={assessmentStyles.scrollView}>
          <View style={[assessmentStyles.scrollViewContent, scrollStyle]}>
            {icon && <SvgXml xml={icon} />}
            {children}
          </View>
        </ScrollView>
        <View style={assessmentStyles.footer}>{footer}</View>
      </ImageBackground>
    </SafeAreaView>
  );
};

export const assessmentStyles = StyleSheet.create({
  container: {
    flex: 1,
    borderTopWidth: 0,
  },
  backgroundImage: {
    flex: 1,
    resizeMode: 'cover',
    justifyContent: 'center',
    borderTopWidth: 0,
  },
  scrollView: {
    flex: 1,
  },
  scrollViewContent: {
    padding: 20,
    borderTopWidth: 0,
  },
  footer: {
    padding: 20,
  },
});
