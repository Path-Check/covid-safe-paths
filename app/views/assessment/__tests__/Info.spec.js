import { render } from '@testing-library/react-native';
import React from 'react';
import { View } from 'react-native';

import { Info } from '../Info';

test('base', () => {
  const { asJSON, queryByTestId } = render(
    <Info image={{ uri: 'image.png' }} title='End' />,
  );
  expect(asJSON()).toMatchSnapshot();
  expect(queryByTestId('cta')).toBe(null);
});

test('children', () => {
  const { getByTestId } = render(
    <Info image={{ uri: 'image.png' }} title='End'>
      <View testID='children' />
    </Info>,
  );
  expect(getByTestId('children')).toBeTruthy();
});

test('footer', () => {
  const { getByTestId } = render(
    <Info
      footer={<View testID='footer' />}
      image={{ uri: 'image.png' }}
      title='End'
    />,
  );
  expect(getByTestId('footer')).toBeTruthy();
});
