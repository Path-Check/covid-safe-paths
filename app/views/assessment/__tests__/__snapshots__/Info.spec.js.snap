// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <SafeAreaView
    style={
      Object {
        "borderTopWidth": 0,
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "borderTopWidth": 0,
          "flex": 1,
          "justifyContent": "center",
          "resizeMode": "cover",
        }
      }
    >
      <Image
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <ScrollView
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderTopWidth": 0,
                "padding": 20,
              },
              undefined,
            ]
          }
        />
      </ScrollView>
      <View
        style={
          Object {
            "padding": 20,
          }
        }
      />
    </View>
  </SafeAreaView>
</View>
`;
